config/src/main/java/org/springframework/security/config/annotation/method/configuration/SecuredMethodSecurityConfiguration.java

```diff
@@ -48,8 +52,13 @@ final class SecuredMethodSecurityConfiguration {
 	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
 	static MethodInterceptor securedAuthorizationMethodInterceptor(
 			ObjectProvider<SecurityContextHolderStrategy> strategyProvider,
-			ObjectProvider<ObservationRegistry> registryProvider) {
+			ObjectProvider<ObservationRegistry> registryProvider, ApplicationContext context) {
```

Comment:
Is there a reason to use `ApplicationContext` instead of `ObjectProvider`? I'd prefer to avoid future maintenance questions, if possible, by having this feature follow the same patterns other features.

It also makes this beans dependencies clearer.

Reply:
It also may ultimately be less code to maintain:

```java
RoleHierarchy roleHierarchy = roleHierarchyProvider.getIfAvailable(NullRoleHierarchy::new);
```